www/api/search

```diff
@@ -66,11 +67,240 @@ soon as they're available in the UI.
+16 random letters and numbers.  This format was chosen because (a) it
+reliably yields gibberish strings without word breaks, which makes it
+an excellent key for a generic Web text search, (b) it's
+<i>effectively</i> a sort of UUID, since the range is so large
+(there are about 8 trillion trillion possible combinations) that
+collisions with similarly formatted ID keys in other databases will
+essentially never happen, and (c) it doesn't <i>look</i> like a
+standard UUID, which might have been somewhat confusing given that the
+Babel spec uses standard UUIDs for IFIDs.
+
+<p>(It might seem strange that IFDB doesn't just use Babel IFIDs as
+primary identifiers, but there are good reasons: (a) Babel only
+applies IFIDs to games, whereas IFDB needs primary identifiers for
+other kinds of entities, such as recommended lists, user profiles, and
+polls; and (b) a single game can have multiple Babel IFIDs, so even
+for games there'd still be a need in the database structure for a
+single unique identifier per game.)
+
+<h3>Games</h3>
+
+<p>
+<pre>
+{
+  "games": [
+    {
+      "tuid": "<i>TUID</i>",
+      "title": "<i>Title of the game</i>",
+      "link": "<i>URL to game's main listing page</i>",
+      "author": "<i>Author(s)</i>",
+      "hasCoverArt": <i>true</i>,
+      "devsys": "<i>Development system used to create this game</i>",
+      "published": {
+        "machine": "<i>Publication date in YYYY or YYYY-MM-DD format</i>",
+        "printable": "<i>Publication date in English format</i>"
+      },
+      "averageRating": <i>Raw "star" rating average</i>,
+      "numRatings": <i>Number of ratings in the average (note that reviews can choose to "omit" their ratings from the average)</i>,
+      "starRating": <i>Rounded "star" rating average</i>,
+      "starSort": <i>Star rating for sorting; see below</i>,
+      "coverArtLink": "<i>URL to cover art</i>"
+    }
+  ]
+}
+</pre>
+</p>
+
+<p>The "star" rating average is given in three forms.  The "raw" average
+is the arithmetic average of the ratings, as a decimal floating point
+value (e.g., "3.3333").  The "rounded" average is the average value
+rounded to the nearest half-star (e.g., "3.5").  IFDB itself generally
+displays the rounded value, since it uses a graphical image with
+half-star resolution.
+
+<p>We determine the games with the highest ratings using
+<a href="https://www.evanmiller.org/ranking-items-with-star-ratings.html">Evan
+Miller's formula</a>, which we call "starSort." We used to sort by average
+rating, but this would tend to rank games with just one perfect 5-star rating
+above games with dozens of 5-star ratings and a few 4-star ratings.
+Evan Miller's formula sorts by our confidence in the game, by adding five
+"fake" ratings to the average (one 1-star, one 2-star, one 3-star, one 4-star,
+and one 5-star rating) and subtracting the standard deviation from the result.
+
+<p>The hasCoverArt value is simply "true" or "false", indicating whether
```

Comment:
This was left in for compatibility with ifiction.

It's enough to have/not have the cover art URL as an indication that a cover exists. Maybe it can be removed from the JSON output.

Reply:
I'm not going to bother removing it from the JSON output for now (it's easily ignored), but I agree that it streamlines the documentation not to mention `hasCoverArt` or `hasPicture`.